### DISCUSSION TITLE
Suggestion: Add PWM method as option for parameter estimation

### DISCUSSION THREAD

**Discussion by robynlo** [georgebv/pyextremes#32] [General]
Thank you for creating this nice user-friendly tool. 

According to a report by Caires (2011, JCOMM Technical Report No. 57) the method of probability weighted moments (PWM) typically performs better than the more sophisticated and flexible ML method in the estimation of the GPD and GEV parameters for ordinary extreme value analyses, such as those of wave data (See report at https://library.wmo.int/doc_num.php?explnum_id=7446).  It would be very nice to be able to use your Code and select PWM as an option to compare the results from this method to the currently supported ML and MCMC.

**Comment by laraibafroz** [comment]
Yes, please. PWM and OLS methods would be really appreciated.

**Comment by shenxiangzhuang** [comment]
Just for reference, I have done some tests in our industry time series data with [SPOT Algorithm](https://hal.science/hal-01640325/file/siffer_kdd_17.pdf), which is a time series anomaly detection algorithm supported by EVT. In SPOT, I do find that MOM and ML estimation methods work both in general, but MOM would break in some cases where ML method also works badly...

In production environment, we choose to use MOM at most due to it's simplicity and efficiency(in cpu). Another reason is that ML method doesn't work well too(in some cases). 

I don't know if PWM will solve my problem or not, but I'll try to learn about this.

**Comment by georgebv** [comment]
All of these sound like good ideas. One would need to implement a `Model` class from this https://github.com/georgebv/pyextremes/blob/master/src/pyextremes/models/model_base.py with the following:

1. `name` attribute
2. `fit` method
3. `get_return_value` method

A good start would be a PR with implementation of one of these methods. It can be implemented for a subset of `scipy` distributions (GEVD and GPD are obvious choices). If someone gets this going, I will gladly review and provide guidance if needed.
